test/tools/libtesteth/ImportTest.cpp

```diff
@@ -241,10 +263,62 @@ bytes ImportTest::executeTest(bool _isFilling)
             if(opt.trValueIndex != -1 && opt.trValueIndex != tr.valInd)
                 continue;
 
-            std::tie(tr.postState, tr.output, tr.changeLog) =
-                    executeTransaction(net, *m_envInfo, m_statePre, tr.transaction);
-            tr.netId = net;
-            transactionResults.push_back(tr);
+            // Skip transaction if there is no expect section for this transaction (noindex data,
+            // gas, val, network)
+            bool found = false;
+            json_spirit::mArray expects;
+            if (_isFilling)
+                expects = m_testInputObject.at("expect").get_array();
+            else
+                expects = m_testInputObject.at("post").get_obj().at(netIdToString(net)).get_array();
+
+            for (auto const& exp : expects)
+            {
+                json_spirit::mObject const& expSection = exp.get_obj();
+
+                if (_isFilling)
+                {
+                    set<string> networkSet;
+                    parseJsonStrValueIntoSet(expSection.at("network"), networkSet);
+                    networkSet = test::translateNetworks(networkSet);
+                    if (!networkSet.count(netIdToString(net)))
+                        continue;
+                }
+
+                vector<int> d;
+                vector<int> g;
+                vector<int> v;
+                BOOST_REQUIRE_MESSAGE(expSection.at("indexes").type() == jsonVType::obj_type,
+                    "indexes field expected to be json Object!");
+                json_spirit::mObject const& indexes = expSection.at("indexes").get_obj();
+                parseJsonIntValueIntoVector(indexes.at("data"), d);
+                parseJsonIntValueIntoVector(indexes.at("gas"), g);
+                parseJsonIntValueIntoVector(indexes.at("value"), v);
+                BOOST_CHECK_MESSAGE(d.size() > 0 && g.size() > 0 && v.size() > 0,
+                    TestOutputHelper::get().testName() + " Indexes arrays not set!");
+                if ((inArray(d, tr.dataInd) || inArray(d, -1)) &&
+                    (inArray(g, tr.gasInd) || inArray(g, -1)) &&
+                    (inArray(v, tr.valInd) || inArray(v, -1)))
+                    found = true;
+                if (found)
+                    break;
+            }
+            if (!found)
+                continue;
+
+
+            transactionToExecute tr2 = tr;
```

Comment:
call it `transactuinResult`
(weird that the same type represents both the transaction and transaction result)

Reply:
looking at this code after one year I am surprised how tangled it is and how it is hard to make any changes. I really hope to drop testeth support soon.

Reply:
you're welcome to review retesteth code to prevent this entanglement from happening. 
that's voluntary